shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/presentation/StepQuizFeature.kt

```diff
@@ -69,29 +68,16 @@ interface StepQuizFeature {
         data class SyncReply(val reply: Reply) : Message
 
         /**
-         * Request user permission
+         * Reset code
          */
-        data class RequestUserPermission(val userPermissionRequest: StepQuizUserPermissionRequest) : Message
-        data class RequestUserPermissionResult(
-            val userPermissionRequest: StepQuizUserPermissionRequest,
-            val isGranted: Boolean
-        ) : Message
-
-        /**
-         * Show problem of day solve modal
-         */
-
-        data class ShowProblemOfDaySolvedModal(val earnedGemsText: String) : Message
-
-        object ProblemOfDaySolvedModalGoBackClicked : Message
+        object RequestResetCode : Message
```

Comment:
Is it assumed to be sent from platform UI?
In my opinion, the name is quite confusing. It is the same as the `Action.ViewAcion.RequestResetCode`.
What about ResetCodeClicked?

Reply:
In addition, in my opinion, the name is quite confusing. It is the same as the `Action.ViewAcion.RequestResetCode`.What about ResetCodeClicked?

Reply:
`StepQuizFeature.Message.RequestResetCode` is unused -> can be safely deleted

Reply:
We are requesting permission for the code reset when user wants to create new submission (via retry button on Step screen or on fullscreen code editor via reset code button):
```kotlin
is Message.CreateAttemptClicked ->
    if (state is State.AttemptLoaded) {
        if (state.step.block.name == BlockName.CODE || state.step.block.name == BlockName.SQL) {
            state to setOf(Action.ViewAction.RequestResetCode)
        } else {
            State.AttemptLoading(oldState = state) to setOf(
                Action.CreateAttempt(
                    message.step,
                    state.attempt,
                    state.submissionState,
                    message.shouldResetReply
                )
            )
        }
    } else {
        null
    }
```